glotaran/deprecation/deprecation_utils.py

```diff
@@ -0,0 +1,504 @@
+"""Helper functions to give deprecation warnings."""
+
+from __future__ import annotations
+
+import os
+import sys
+from functools import wraps
+from importlib import import_module
+from importlib.metadata import distribution
+from types import ModuleType
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Callable
+from typing import TypeVar
+from typing import cast
+from warnings import warn
+
+DecoratedCallable = TypeVar(
+    "DecoratedCallable", bound=Callable[..., Any]
+)  # decorated function or class
+
+if TYPE_CHECKING:
+    from typing import Sequence
+
+
+class OverDueDeprecation(Exception):
+    """Error thrown when a deprecation should have been removed.
+
+    See Also
+    --------
+    deprecate
+    warn_deprecated
+    deprecate_module_attribute
+    deprecate_submodule
+    """
+
+    pass
```

Comment:
Is this pass here needed? codacy [doesn't think so](https://app.codacy.com/gh/glotaran/pyglotaran/pullRequest?prid=7203405).

Reply:
Ohhh nice this was new to me, looks like when you define a docstring you can skip the `pass`, which else is required.